### DISCUSSION TITLE
analyze the build github action environment for macos 11.0 M1

### DISCUSSION THREAD

**Discussion by MrBitBucket** [pypa/cibuildwheel#1045] [General]
macos 10.9 actions are good and at delocate time I see this

`+ delocate-listdeps /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/built_wheel/reportlab-3.6.8-cp39-cp39-macosx_10_9_x86_64.whl && delocate-wheel --require-archs x86_64 -w /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/repaired_wheel /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/built_wheel/reportlab-3.6.8-cp39-cp39-macosx_10_9_x86_64.whl
  /usr/local/Cellar/freetype/2.11.1/lib/libfreetype.6.dylib`

I assume Cellar here refers to the homebrew cellar. 

The arm64 version doesn't mention freetype

`  + delocate-listdeps /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/built_wheel/reportlab-3.6.8-cp39-cp39-macosx_11_0_arm64.whl && delocate-wheel --require-archs arm64 -w /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/repaired_wheel /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/cibuildwheel4vaswxw1/built_wheel/reportlab-3.6.8-cp39-cp39-macosx_11_0_arm64.whl`

and I get runtime errors from users with M1 when they try to use TTF files (handled by freetype).

Can I assume that the cibuildwheel macos 10.9 environment has homebrew?
Can I assume the same for macos 11.0?

The reportlab setup.py has a very old search for include/lib files and it certainly doesn't mention the Cellar folder.
Is there a good example for how to find the right files for freetype2? I am unfamiliar with macos and without any hardware I rely on an aging sierra VM.

Failing anything explicit how would I go about analyzing the github action environment.

**Comment by MrBitBucket** [comment]
I checked a repeat build and see that indeed brew reports freetype-2.11.1 is already installed. I see this coming from the build

` ld: warning: ignoring file /usr/local/lib/libfreetype.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
    installing to build/bdist.macosx-11.0-arm64/wheel` which is obviously wrong. Is this something I can fix?

**Reply by Czaki** [reply]
At first your build process should build all libraries from scratch, not use brew one, because homebrew build libraries against current system, not macos 10.9. 

To solve this problem you need to build libfreetype for ARM and make it accessible for dealocate.

**Reply by henryiii** [reply]
Yes, homebrew doesn't build against older versions of macOS, or produce universal binaries (there's no need to), but that makes it pretty useless for building wheels. GHA (and most macOS CI's) have homebrew. You can `brew unlink freetype` first if it's tripped up by something installed by default.

**Reply by MrBitBucket** [reply]
I don't have a problem with 10.9/10.15. I have problem with 11.0 and the M1 arm64 build. The error says that although the build is intended for arm64 it will ignore the arm64 dylib and instead try to link against x86_64 lib which of course I don't have. The link does not fail though, but the delocate doesn't include any libfreetype/libpng dylibs. So are you both saying that homebrew doesn't support arm64?
